data/private

```diff
@@ -119,7 +119,7 @@ b.e.f.ip6.arpa
 127.100.in-addr.arpa
 
 # Dotless domains
-regexp:^[^.]+$
+regexp:^[A-Za-z0-9]([A-Za-z0-9-]{1,61}[A-Za-z0-9]|[A-Za-z0-9]|$)$
```

Comment:
In my opinion, I suggest using a shorter regex to match the dotless domain. Although your syntax is correct, I don't think some parts are necessary. For example, First, I think you don't need to limit the domain to 63 characters because this kind of domain does not exist. Nobody will send the domain which is over 63 characters to v2ray. Second, the domain start or ends with single or multiple dashes does not exist. In other words, your regex is trying to prevent the cases which are impossible.
I think your regex just needs to process the situations which are possible. e.g.:regexp:^[-0-9A-Za-z]+$
Of course, it is just my personal opinion, so if you don't want to modify it. That's fine.

Reply:
P.S. If you don't want to modify it, just press "Resolve conversation".

Reply:
> Nobody will send the domain which is over 63 characters to v2ray. Second, the domain start or ends with single or multiple dashes does not exist. In other words, your regex is trying to prevent the cases which are impossible.

We can say the same about https://github.com/v2fly/domain-list-community/issues/1395 issue. Nobody will send IP address as SNI header, but errors happens as we can see.

But I agree with `^[-0-9A-Za-z]+$` and I think that it is enough.

Also, I tried to use hostname with trailing dash in linux environment and looks like it works:
```
# echo '127.0.0.1 test-hostname-' >> /etc/hosts
# ping test-hostname-
PING test-hostname- (127.0.0.1) 56(84) bytes of data.
64 bytes from localhost.localdomain (127.0.0.1): icmp_seq=1 ttl=64 time=0.086 ms
```

So not everyone follows these rules.

Reply:
>We can say the same about https://github.com/v2fly/domain-list-community/issues/1395 issue. Nobody will send IP address as SNI header, but errors happens as we can see.

Well, that is impossible in v2ray or xray, but it happens in sing-box, so this problem needs to solve.

And thank you for accepting my suggestion.